third_party/java/arcs/build_defs/internal/manifest.bzl

```diff
@@ -1,16 +1,31 @@
 """Arcs manifest bundling rules."""
 
+load(":run_test.bzl", "run_test")
 load(":tools.oss.bzl", "arcs_tool_manifest2proto", "arcs_tool_recipe2plan")
 load(":util.bzl", "replace_arcs_suffix")
 
-def arcs_manifest(name, srcs, deps = [], visibility = None):
+def arcs_manifest(
```

Comment:
Instead of changing the `arcs_manifest` API, can we refactor this into an internal helper (say `arcs_manifest_helper`) and create specific rules like `arcs_manifest` and `policy_test` that call `arcs_manifest_helper`. I think that will be more readable and we won't accidentally include a policy_test as a dependency.

Reply:
Part of the reason I went for this approach was to make it easier to create the policy test (or harder to forget to create it).

If we require developers to do all of the following for every manifest file they write, there will be a lot of occurrences where they forget to invoke one or multiple of the rules:

```
arcs_manifest(
  name = "foo",
  srcs = ["foo.arcs"],
)

arcs_manifest_proto(
  name = "foo_proto",
  srcs = ["foo.arcs"],
)

arcs_policy_test(
  name = "foo_policy_test",
  manifest_proto = ":foo_proto",
)
```

It seems like a much nicer API to just do this:

```
arcs_manifest(
  name = "foo",
  srcs = ["foo.arcs"],
)
```

where everything gets created for you automatically. 

As for your last point ("we won't accidentally include a policy_test as a dependency"), that's already sorted for us. You can't depend on test rules from non-test rules, bazel/blaze won't let you.

Reply:
I agree that we should not have the developers multiple rules. I wasn't suggesting that. My comment was more about readability and naming.

I might have misunderstood what these changes do.  Consider the following rule (say "//some/directory:foo"):
```
arcs_manifest(
  name = "foo",
  srcs = ["foo.arcs"],
  manifest_proto = True,
  policy_test = True,
  policy_options = "options.textproto",
)
```

I *thought* that this would generate both a `arcs_manifest_proto` rule and a *test-only* `arcs_verify_policy_tool` rule. However, that is not the case. It actually generates a `arcs_manifest_proto`  and invokes policy verifier. Is that right?

Reply:
> I thought that this would generate both a arcs_manifest_proto rule and a test-only arcs_verify_policy_tool rule.

That's right. It generates all of the following:

```
filegroup(name = "foo", ...)
arcs_manifest_proto(name = "foo_manifest_proto", ...)
arcs_tool_verify_policy(name = "foo_policy_test", ...)
```

If you dep on `:foo` then you'll pull in the filegroup (i.e. just the .arcs manifest file). But those other targets `:foo_manifest_proto` and `:foo_policy_test` will also be generated, and you can use them if you want. You can use the manifest proto via either `:foo_manifest_proto` or via its output file `foo.bin.pb`